src/ConversationData.res

```diff
@@ -204,94 +152,99 @@ let fetchMessages = (~conversationId: int, ~immobilieId: int): Js.Promise.t<arra
      |> ignore
    );
  */
-let postReply = (conversation: conversation, msg: string, attachments: array<string>): Js.Promise.t<
-  message,
-> => {
-  let dict = Js.Dict.empty()
-  Js.Dict.set(dict, "message", Js.Json.string(msg))
-  Js.Dict.set(dict, "attachments", Js.Json.stringArray(attachments))
-  Fetch.fetchWithInit(
+let postReply = async (
+  conversation: conversation,
+  msg: string,
+  attachments: array<string>,
+): message => {
+  open Fetch
+
+  let dict = Dict.fromArray([
+    ("message", Js.Json.string(msg)),
+    ("attachments", Js.Json.stringArray(attachments)),
+  ])
+
+  let data = await fetch(
     apiBaseUrl ++
     ("/anfragen/immobilie/" ++
     (Belt.Int.toString(conversation.immobilie_id) ++
     ("/conversations/" ++
     (Belt.Int.toString(conversation.id) ++ "/reply")))),
-    Fetch.RequestInit.make(
-      ~credentials=Include,
-      ~method_=Post,
-      ~body=Fetch.BodyInit.make(Js.Json.stringify(Js.Json.object_(dict))),
-      (),
-    ),
-  )
-  ->Promise2.then(Fetch.Response.json)
-  ->Promise2.thenResolve(json => json->Decode.single_message)
+    {
+      credentials: #"include",
+      method: #POST,
+      body: dict->Js.Json.stringifyAny->Option.getExn->Body.string,
```

Comment:
I am not entirely happy with this. At some point one could use the new variant representation to easily represent Js.Json.t and then just construct proper Json objects yourself (more on that in the blog post about variants linked above).

For now this will suffice... we are creating a dictionary from a bunch of key/value pairs, then unsafely coerce it into a JSON and then to a string for the POST fetch.